oop/src/PurchaseOrders/use-cases/CreatePO.usecase.ts

```diff
@@ -1,12 +1,31 @@
 import { UseCase } from "../../lib/UseCase.base";
-import { IPORepository } from "../domain/IPORepository";
-import { PurchaseOrder } from "../domain/PurchaseOrder";
+import { UUID, createUuid } from "../../utilities/uuid";
+import { IPORepository } from "../domain/PurchaseOrder.repo-interface";
+import { PurchaseOrder } from "../domain/PurchaseOrder.entity";
+import { CreatePOLineItemDTO } from "../infrastructure/dto/CreatePO.dto";
+import { PurchaseOrderLineItem } from "../domain/PurchaseOrderLineItem.entity";
 
-type CreatePOProps = {};
+type CreatePOProps = {
+  id: UUID;
+  lineItems?: CreatePOLineItemDTO[];
+};
 export class CreatePO implements UseCase<CreatePOProps> {
   constructor(private repo: IPORepository) {}
-  async execute() {
-    const purchaseOrder = new PurchaseOrder();
+  async execute({ id, lineItems }: CreatePOProps) {
+    const purchaseOrder = new PurchaseOrder({
+      id,
+      lineItems: lineItems?.map(
+        (lineItem) =>
+          new PurchaseOrderLineItem({
+            id: createUuid(),
```

Comment:
Don't create the id here. Instead use the value from the passed in line item DTO.